frontend/src/views/Game.tsx

```diff
@@ -113,8 +114,7 @@ function GamePage() {
       setStateFromGame(updatedGame);
 
       // Check if end game.
-      // TODO: after deconstructing game, move this to useEffect on timeUp
-      if (updatedGame.gameTimer.timeUp) {
+      if (updatedGame.gameTimer.timeUp || updatedGame.allSolved) {
         history.push('/game/results', {
           game: updatedGame,
         });
```

Comment:
Suggestion: save `allSolved` and `timeUp` state variables in the `setStateFromGame` function and move to a separate useEffect block (that way the redirect will occur even if refreshing at the perfect time)

Reply:
Good suggestion! Done ✅ 

I did not pass any state variables through `history.push`, as I saw you handled this logic in #97 and there's no use for it in this PR.